pkg/controller/stackconfigpolicy/stackconfigpolicy.go

```diff
@@ -0,0 +1,489 @@
+// It processes all provided policies, filtering those that target the Elasticsearch cluster, and merges them
+// in order of their weight (lowest to highest). Policies with the same weight are flagged as conflicts.
+// Returns an esPolicyConfig containing the merged configuration and any error occurred during merging.
+func getPolicyConfigForElasticsearch(es *esv1.Elasticsearch, allPolicies []policyv1alpha1.StackConfigPolicy, params operator.Parameters) (*esPolicyConfig, error) {
+	esPolicy := esPolicyConfig{
+		PoliciesWithConflictErrors: make(map[types.NamespacedName]error),
+	}
+	if len(allPolicies) == 0 {
+		return &esPolicy, nil
+	}
+
+	// Group policies by weight
+	var weights []int32
+	weightKeyStackPolicies := make(map[int32][]*policyv1alpha1.StackConfigPolicy)
+	for _, p := range allPolicies {
+		isRef, err := doesPolicyRefsObject(&p, es, params.OperatorNamespace)
+		if err != nil {
+			return nil, err
+		}
+		if !isRef {
+			// policy does not target the given Elasticsearch
+			continue
+		}
+
+		if _, exists := weightKeyStackPolicies[p.Spec.Weight]; !exists {
+			weights = append(weights, p.Spec.Weight)
+		}
+
+		weightKeyStackPolicies[p.Spec.Weight] = append(weightKeyStackPolicies[p.Spec.Weight], &p)
+		esPolicy.PoliciesRefs = append(esPolicy.PoliciesRefs, p)
+	}
+
+	if len(esPolicy.PoliciesRefs) == 1 {
+		// Since we have only one policy avoid merging (including canonicalise)
+		// and thus avoid any reconciliation storm caused by unnecessary
+		// secret changes
+		esConfigPolicy := esPolicy.PoliciesRefs[0].Spec.Elasticsearch.DeepCopy()
+		esPolicy.Spec = *esConfigPolicy
+		return &esPolicy, nil
+	}
+
+	// Process policies in order of weight (lowest first)
+	slices.Sort(weights)
+
+	// Reverse the weights so that we process policies in order of weight (highest first)
+	slices.Reverse(weights)
+
+	var previouslyAppliedPolicy *policyv1alpha1.StackConfigPolicy
+	for _, weight := range weights {
+		policiesWithSameWeight := weightKeyStackPolicies[weight]
+		if len(policiesWithSameWeight) > 1 {
+			// Multiple policies with the same weight - this is a conflict
+			conflictErr := getPolicyConflictError(policiesWithSameWeight, weight)
+			for _, p := range policiesWithSameWeight {
+				esPolicy.PoliciesWithConflictErrors[k8s.ExtractNamespacedName(p)] = conflictErr
+			}
+			return &esPolicy, conflictErr
+		}
+		policy := policiesWithSameWeight[0]
+		// Merge the single policy at this weight level
+		if err := mergeElasticsearchConfig(&esPolicy.Spec, policy.Spec.Elasticsearch); err != nil {
+			if errors.Is(err, errMergeConflict) {
+				policyNsn := k8s.ExtractNamespacedName(policy)
+				esPolicy.PoliciesWithConflictErrors[policyNsn] = err
+
+				if previouslyAppliedPolicy != nil {
+					previouslyAppliedPolicyNsn := k8s.ExtractNamespacedName(previouslyAppliedPolicy)
+					esPolicy.PoliciesWithConflictErrors[previouslyAppliedPolicyNsn] = err
+				}
```

Comment:
I am not sure I understand this code. I assme it is to track with which policy we have a conflict? But what if we have three: 

Policy A (weight 10): secretMounts with mountPath "/data"
Policy B (weight 5):  clusterSettings only (no secretMounts)
Policy C (weight 0):  secretMounts with mountPath "/data" ← conflicts with A!

Would you not incorrectly track a conflict with B here?